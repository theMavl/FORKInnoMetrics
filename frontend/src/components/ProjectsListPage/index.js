import React from 'react'
import {connect} from 'react-redux'
import {getProjectsRequest} from '../../store/projects/actionCreators'
import PageTemplate from '../PageTemplate'
import styles from './style.css'

class _ProjectsListPage extends React.Component {
  componentDidMount(){
    this.props.getProjects()
  }
  render() {
    return (
      <PageTemplate title='Projects'
                    // restHeader={<Button name='New project'
                    //                     icon='add'
                    //                     styleType='action'
                    // />}
      >
        {this.props.activeRequest ? 'Loading ... ' :
          (this.props.projects.length === 0 ?
            'There is nothing to show yet' :
            <div className={styles.projectsList}>
              {this.props.projects.map(p => (
                <div className={styles.project}
                     key={p.name}
                     onClick={() => this.props.history.push(`/projects/${p.name}`)}
                >
                  <span className={styles.projectName}>{p.name}</span>
                </div>
              ))}
            </div>
        )}
      </PageTemplate>

    )
  }
}

const ProjectsListPage = connect(
  (state) => ({
    projects: state.projects.projects,
    activeRequest: state.projects.activeRequest
  }),

  (dispatch) => ({
    getProjects: () => dispatch(getProjectsRequest())
  })
)(_ProjectsListPage)

export default ProjectsListPage