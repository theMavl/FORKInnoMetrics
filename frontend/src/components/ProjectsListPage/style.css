.project {
    min-height: var(--cell-size);
    width: 100%;
    padding: 0 calc(var(--cell-size) / 4);
    box-shadow: 0 0 5px 0px rgba(100, 100, 100, 0.15);
    margin-bottom: 15px;

    display: flex;
    align-items: center;

    cursor: pointer;
}

.projectName {
    font-size: larger;
}