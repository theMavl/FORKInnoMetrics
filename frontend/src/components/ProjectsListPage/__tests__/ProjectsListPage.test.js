import React from 'react'
import createStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'
import TestRenderer from 'react-test-renderer'
import ProjectsListPage from '../index'

describe('Test ProjectsListPage component', () => {

  it('Test render ProjectsListPage component', () => {

    const mockStore = createStore([thunk])
    const store = mockStore({
      projects: {
        projects: [{name: 'proj 1'}, {name: 'proj 2'}, {name: 'proj 3'}],
        activeRequest: false
      }
    })

    const instance = TestRenderer.create(
      <Provider store={store}>
        <ProjectsListPage />
      </Provider>
    )

    expect(instance.toJSON()).toMatchSnapshot()
  })

})