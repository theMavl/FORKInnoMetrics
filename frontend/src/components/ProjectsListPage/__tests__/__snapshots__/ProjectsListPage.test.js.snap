// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProjectsListPage component Test render ProjectsListPage component 1`] = `
<article>
  <div>
    <span>
      <h1>
        Projects
      </h1>
    </span>
  </div>
  <div>
    <div>
      <div
        onClick={[Function]}
      >
        <span>
          proj 1
        </span>
      </div>
      <div
        onClick={[Function]}
      >
        <span>
          proj 2
        </span>
      </div>
      <div
        onClick={[Function]}
      >
        <span>
          proj 3
        </span>
      </div>
    </div>
  </div>
</article>
`;
